pkg/kubectl/cmd/version.go

```diff
@@ -46,36 +53,66 @@ func NewCmdVersion(f cmdutil.Factory, out io.Writer) *cobra.Command {
 	}
 	cmd.Flags().BoolP("client", "c", false, "Client version only (no server required).")
 	cmd.Flags().BoolP("short", "", false, "Print just the version number.")
+	cmd.Flags().String("output", "", "output format, options available are yaml and json")
 	cmd.Flags().MarkShorthandDeprecated("client", "please use --client instead.")
 	return cmd
 }
 
 func RunVersion(f cmdutil.Factory, out io.Writer, cmd *cobra.Command) error {
-	v := fmt.Sprintf("%#v", version.Get())
+	vo := VersionObj{nil, nil}
+
+	cvg := version.Get()
 	if cmdutil.GetFlagBool(cmd, "short") {
```

Comment:
I disagree, regardless of the consumer we should respect the expectations of the flag.